go/tools/generate_test_main.go

```diff
@@ -82,33 +79,24 @@ func main() {
 		}
 	}
 
-{{ if .Version.go1_8 }}
+{{if eq .Version "go1.8"}}
```

Comment:
Yes, I could have done this better. A lot of things in the `testing` package are excluded from Go 1 compatibility guidelines, so I expect to have to fix this tool after every Go release. However, it shouldn't break automatically if the upstream API *doesn't* change.